src/components/Cell/index.js

```diff
@@ -24,32 +22,14 @@ import CellStyle from './styles/CellStyle'
  * If given less breakpoints than in the theme, the last one is used.
  * If given more breakpoints than in the theme, the excess ones are ignored.
  */
-class Cell extends React.Component {
-  state = {
-    cols: [],
-  }
-
-  componentWillMount() {
-    this.state.cols = clone(this.props.cols)
-  }
-
-  componentWillReceiveProps(nextProps) {
-    if (this.props.cols !== nextProps.cols) {
-      this.setState({ cols: clone(nextProps.cols) })
-    }
-  }
-
-  render() {
-    const { children, className } = this.props
-
-    const { cols } = this.state
-
-    return (
-      <CellStyle className={className} cols={cols}>
-        {children}
-      </CellStyle>
-    )
-  }
+const Cell = ({ children, cols: propsCols, className }) => {
+  const [cols] = useState(propsCols)
```

Comment:
The useState call here is not the same as the code as it was before.
The code before was initializing the state with an empty array, then was cloning the props into the state (?) and then, each time the `cols` prop changed, was cloning it again into the state. Here, you're not cloning anything, you're just storing the prop into the state **at the first render**. The first parameter of useState is the initial value.
So, if you really want to clone the prop, you need to do :
```
const Cell = ({ children, cols: propsCols, className }) => {
  const [cols, setCols] = useState(propsCols)
  useEffect(() => {
    setCols(clone(cols))
  }, [propsCols])
  return (
    <CellStyle className={className} cols={cols}>
      {children}
    </CellStyle>
  )
}
```
But I think the problem here is not your change. The problem is : why shoul we have to clone the props, whereas the props should never be modified and all side effects on the props should be prevented ?
The real code should be :
```
const Cell = ({ children, cols, className }) => {
return (
    <CellStyle className={className} cols={cols}>
      {children}
    </CellStyle>
  )
}
```

Reply:
Should be good now :) No need for it.